astropy/coordinates/sky_coordinate.py

```diff
@@ -484,6 +487,94 @@ def transform_to(self, frame, merge_attributes=True):
             frame_kwargs.pop(attr)
         return self.__class__(new_coord, **frame_kwargs)
 
+    def apply_space_motion(self, new_obstime=None, dt=None):
+        """
+        Update the position of the source represented by this coordinate object
+        to a new specified time using the velocity and assuming linear motion.
+
+        This is sometimes referred to as an "epoch transformation."
+
+        Parameters
+        ----------
+        new_obstime : `~astropy.time.Time`, optional
+            The time at which to evolve the position to.
+        dt : `~astropy.units.Quantity`, `~astropy.time.TimeDelta`, optional
+            An amount of time to evolve the position of the source.
+        """
+
+        if (new_obstime is None and dt is None or
+                new_obstime is not None and dt is not None):
+            raise ValueError("You must specify one of `new_obstime` or `dt`, "
+                             "but not both.")
+
+        # Validate that we have velocity info
+        if 's' not in self.frame.data.differentials:
+            raise ValueError('SkyCoord requires velocity data to evolve the '
+                             'position.')
+
+        if 'obstime' in self.frame.frame_attributes:
+            raise NotImplementedError("Updating the coordinates in a frame "
+                                      "with explicit time dependence is "
+                                      "currently not supported. If you would "
+                                      "like this functionality, please open an "
+                                      "issue on github:\n"
+                                      "https://github.com/astropy/astropy")
+
+        if new_obstime is not None and self.obstime is None:
+            # If no obstime is already on this object, raise an error if a new
+            # obstime is passed: we need to know the time / epoch at which the
+            # the position / velocity were measured initially
+            raise ValueError('This object has no associated `obstime`. '
+                             'apply_space_motion() must receive a time '
+                             'difference, `dt`, and not a new obstime.')
+
+        t1 = self.obstime
+        if t1 is None:
+            # MAGIC NUMBER: if the current SkyCoord object has no obstime,
+            # assume J2000 to do the dt offset.
+            t1 = Time('J2000')
```

Comment:
Is there some reason we *have* to use this magic number?  I'd instead say it's better to *not* assume it and raise an error that says "must have an obstime in a `SkyCoord` to be able to do this kind of transform", rather than assuming J2000...? That's certainly not a natural assumption for at least some frames (e.g. Galactic).

Reply:
I agree with @eteq here - we can always add it later if we find it is really too troublesome without it.

Reply:
After some experimentation I realized this is actually ok, but *only* if it doesn't propogate to setting the final obstime, because then it's only seen bu `starpm`, which only uses these to do the delta-t calculation.  So the actual absolute time is unimportant.